### PR TITLE
pubspec caching

### PR DESCRIPTION
**Pull request by pq**
Before:

image

### CODE REVIEW

File: lib/src/rules/avoid_web_libraries_in_flutter.dart [hunk 1]
```diff
@@ -52,33 +40,82 @@ class AvoidWebLibrariesInFlutter extends LintRule implements NodeLintRule {
...
+}
+
+class Pubspec {
+  static Map<String, Entry> cache = <String, Entry>{};
+
+  static YamlMap forUnit(CompilationUnit unit, LinterContext context) {
+    final filePath = path.join(context.package.root, 'pubspec.yaml');
+    final entry = cache[filePath];
+    if (entry != null) {
+      if (entry.file.modificationStamp != entry.timestamp) {
```
Comment (bwilkerson) [inline lib/src/rules/avoid_web_libraries_in_flutter.dart]: Getting the modification stamp is not as inexpensive as you might hope, and this gets it twice: once here and once in `update`.

File: lib/src/rules/avoid_web_libraries_in_flutter.dart [hunk 2]
```diff
@@ -52,33 +40,82 @@ class AvoidWebLibrariesInFlutter extends LintRule implements NodeLintRule {
...
+    } catch (_) {
+      // Fall-through.
+    }
+    return YamlMap();
+  }
+
+}
+
+class Pubspec {
+  static Map<String, Entry> cache = <String, Entry>{};
```
Comment (bwilkerson) [inline lib/src/rules/avoid_web_libraries_in_flutter.dart]: There's no way to flush the cache, so it is effectively a slow memory leak.

File: lib/src/rules/avoid_web_libraries_in_flutter.dart [hunk 3]
```diff
@@ -2,13 +2,15 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:io';
```
Comment (bwilkerson) [inline lib/src/rules/avoid_web_libraries_in_flutter.dart]: I can't tell where this is being used, but it's somewhat concerning if we really need it.

**Comment by pq** [comment]
Right.  Too many problems with this approach.

Neverminding the leak, a few runs show the benefits are negligible (and under some circumstances actually negative).

Discussion ongoing in #1813 but closing this for now.

**Pull request closed without merging**